webhooks/cluster.go

```diff
@@ -246,14 +254,20 @@ func (webhook *Cluster) validateTopology(ctx context.Context, old, new *clusterv
 				),
 			)
 		}
-	}
-	// Check to see if the ClusterClass referenced in the Cluster currently exists.
-	if err := webhook.Client.Get(ctx, client.ObjectKey{Namespace: new.Namespace, Name: new.Spec.Topology.Class}, &clusterv1.ClusterClass{}); err != nil {
-		allErrs = append(
-			allErrs, field.Invalid(
-				field.NewPath("spec", "topology", "class"),
-				new.Spec.Topology.Class,
-				"ClusterClass could not be found"))
+
+		// If the ClusterClass name has changed (ClusterClass rebase) additional compatibility checks are needed.
+		if old.Spec.Topology.Class != new.Spec.Topology.Class {
+			oldClusterClass := &clusterv1.ClusterClass{}
+			// Check to see if the ClusterClass referenced in the Cluster currently exists.
```

Comment:
this comments seems out of date

Reply:
I'm not sure what seems out of date here - I'll try to clarify the comments.

Reply:
I think it just seems to be outdated because the comment is before the whole block which includes the compatibility check. I think it would be clearer if there is an empty line after l.268
(so that it's clearer that the comment just applies to the get)

Reply:
What I was referring to is the fact that the comment is about "the ClusterClass referenced in the Cluster currently", while instead we are reading the old cluster class

Reply:
I'll clarify that